src/app/organization/entity-data-table-checks/entity-data-table-checks.component.ts

```diff
@@ -0,0 +1,100 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatTableDataSource, MatPaginator, MatSort, MatDialog } from '@angular/material';
+import { ActivatedRoute } from '@angular/router';
+
+/** rxjs Imports */
+import { of } from 'rxjs';
+
+/** Custom Services */
+import { OrganizationService } from '../organization.service';
+
+/** Custom Components */
+import { DeleteDialogComponent } from '../../shared/delete-dialog/delete-dialog.component';
+
+/**
+ * Entity Data Table Checks component.
+ */
+@Component({
+  selector: 'mifosx-entity-data-table-checks',
+  templateUrl: './entity-data-table-checks.component.html',
+  styleUrls: ['./entity-data-table-checks.component.scss']
+})
+export class EntityDataTableChecksComponent implements OnInit {
+
+  /** Entity Data Table Checks data. */
+  entityDataTableChecksData: any;
+  /** Columns to be displayed in entity data table checks table. */
+  displayedColumns: string[] = ['entity', 'productName', 'datatableName', 'status.value', 'systemDefined', 'actions'];
+  /** Data source for entity data table checks table. */
+  dataSource: MatTableDataSource<any>;
+
+  /** Paginator for entity data table checks table. */
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  /** Sorter for entity data table checks table. */
+  @ViewChild(MatSort) sort: MatSort;
+
+  /**
+   * Retrieves the entity data table checks data from `resolve`.
+   * @param {OrganizationService} organizationService Organization Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {MatDialog} dialog Dialog reference.
+   */
+  constructor(private organizationService: OrganizationService,
+              private route: ActivatedRoute,
+              private dialog: MatDialog) {
+    this.route.data.subscribe(( data: { entityDataTableChecks: any }) => {
+      this.entityDataTableChecksData = data.entityDataTableChecks.pageItems;
```

Comment:
@abhaychawla I've implemented it in a slightly new way, as no other components yet have pagination without sorting and filtering. Please tell me what you think about it.

Reply:
I did not use a datasource as I am still needing to use the client side sorting and filtering in MatTableDataSource. I also did not use a resolver, as data will have to be regularly fetched rather than just once.